x-pack/plugins/security_solution/public/common/components/navigation/use_security_solution_navigation/use_navigation_items.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import { APP_ID } from '../../../../../common/constants';
+import { track, METRIC_TYPE, TELEMETRY_EVENT } from '../../../lib/telemetry';
+import { getSearch } from '../helpers';
+import { PrimaryNavigationItemsProps } from './types';
+import { useKibana } from '../../../lib/kibana';
+import { SecurityPageName } from '../../../../app/types';
+import { useFormatUrl } from '../../link_to';
+
+export const usePrimaryNavigationItems = ({
+  filters,
+  navTabs,
+  query,
+  savedQuery,
+  selectedTabId,
+  sourcerer,
+  timeline,
+  timerange,
+}: PrimaryNavigationItemsProps) => {
+  const history = useHistory();
+  const { navigateToApp, getUrlForApp } = useKibana().services.application;
+
+  const navItems = Object.values(navTabs).map((tab) => {
+    const { id, href, name, disabled, pageId } = tab;
+    // Rules of hooks prevents hooks from being called in callbacks as the order of the hooks
+    // should be consistent across every render. The primary nav tabs here are from a constants file
+    // not dynamically generated, so their order will remain consistent across renders.
+    // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
It's probably not worth the effort but would the alternative be to return a callback instead of `navItems` that could be called while iterating over `navTabs`? So the caller of `usePrimaryNavigationItems` would have to do the looping and call the callback returned by `usePrimaryNavigationItems`. Or would that not work?

Reply:
Thanks for the great comment by the way though!

Reply:
Yea, that would work. Actually probably a better route to go to not have to disable any rules of hooks that could cause bugginess in the future if anything changed. Turns out though that I didn't need this code at all as the primary nav doesn't have sub-tabs. Once @semd's changes are in we should be able to get rid of all pageId references and do a final nav cleanup